src/timepicker/TimePicker.ts

```diff
@@ -0,0 +1,356 @@
+	onRequestOptions?(value: string, getOptions: () => TimeUnits[]): void;
+	openOnFocus?: boolean;
+	options?: TimeUnits[];
+	readOnly?: boolean;
+	required?: boolean;
+	start?: string;
+	step?: number;
+	useNativeElement?: boolean;
+	value?: string;
+}
+
+export interface TimeUnits {
+	hour: number;
+	minute?: number;
+	second?: number;
+}
+
+/**
+ * Generate an array of time unit objects from the specified start date to the specified end date.
+ *
+ * @param start    The start time. Defaults to midnight.
+ * @param end      The end time. Defaults to 23:59:59.
+ * @param step     The amount of time in seconds between each step. Defaults to 60.
+ * @return         An array of time unit objects.
+ */
+export function getOptions(start: string = '00:00:00', end: string = '23:59:59', step = 60): TimeUnits[] {
+	const endUnits = parseUnits(end);
+	const startUnits = parseUnits(start);
+	const endTime = getTime(endUnits);
+	const date = new Date();
+	let time = getTime(startUnits, date);
+
+	const i = step * 1000;
+	const options: number[] = [];
+
+	while (time <= endTime) {
+		options.push(time);
+		time += i;
+	}
+
+	return options.map((time: number) => {
+		const date = new Date(time);
+		return {
+			hour: date.getHours(),
+			minute: date.getMinutes(),
+			second: date.getSeconds()
+		};
+	});
+}
+
+/**
+ * @private
+ * Create a numeric timestamp for the specified hour, minute, and second units.
+ *
+ * @param units   An object containing the hours, minutes, and seconds for the time.
+ * @param date    An optional date object.
+ * @return        The timestamp, in milliseconds, according to universal time.
+ */
+function getTime(units: TimeUnits, date = new Date()) {
+	const { hour, minute, second } = units;
+	date.setHours(hour);
+	date.setMinutes(minute as number);
+	date.setSeconds(second as number);
+	date.setMilliseconds(0);
+	return date.getTime();
+}
+
+/**
+ * Convert a standard time string into an object with `hour`, `minute`, and `second` number properties.
+ *
+ * For example, '12:30' is converted to `{ hour: 12, minute: 30, second: 0 }`, and '19:03:27' is converted
+ * to `{ hour: 19, minute: 3, second: 27 }`.
+ *
+ * @param value   A standard time string or an object with `hour`, `minute`, and `second` properties.
+ * @throws Error  For any string not in the format 'HH:mm' or 'HH:mm:ss'.
+ * @return        An object containing `hour`, `second`, and `number` properties.
+ */
+export const parseUnits = (function () {
+	const TIME_PATTERN = /^\d{2}:\d{2}(:\d{2})?$/;
+	return function (value: string | TimeUnits): TimeUnits {
+		if (typeof value === 'string') {
+			if (!TIME_PATTERN.test(value)) {
+				throw new Error('Time strings must be in the format HH:mm or HH:mm:ss');
+			}
+
+			const [ hour, minute, second = 0 ] = value.split(':').map(unit => parseInt(unit, 10));
+			return { hour, minute, second } as TimeUnits;
+		}
+
+		return value;
+	};
+})();
+
+export const TimePickerBase = ThemeableMixin(WidgetBase);
```

Comment:
You will have to export it because it is required to create the typings when doing a `grunt dist`.

An example of the error you would get `src/checkbox/Checkbox.ts(72,39): error TS4020: 'extends' clause of exported class 'Checkbox' has or is using private name 'CheckboxBase'.`